libs/blocks/preflight/checks/preflightApi.js

```diff
@@ -53,3 +55,46 @@ export default {
     runChecks: runChecksSeo,
   },
 };
+
+let preflightResults;
+
+export async function getPreflightResults(url, area, useCache = true) {
+  if (useCache && checks) {
+    preflightResults = await checks;
+    const returnValue = {
+      isViewportTooSmall: isViewportTooSmall(),
+      runChecks: preflightResults,
+    };
+    return returnValue;
+  }
+
+  checks = (async () => {
+    const assets = await Promise.all(runChecksAssets(url, area));
+    const performance = await Promise.all(runChecksPerformance(url, area));
+    const seo = await Promise.all(runChecksSeo({ url, area }));
+    return {
+      assets,
+      performance,
+      seo,
+    };
+  })();
+
+  preflightResults = await checks;
+
+  return {
+    isViewportTooSmall: isViewportTooSmall(),
+    runChecks: preflightResults,
+  };
+}
+
+export function hasPreflightFailures() {
+  if (!preflightResults) return false;
+
+  const runChecks = preflightResults.runChecks || preflightResults;
+  const allResults = [
+    ...(runChecks.assets || []),
+    ...(runChecks.performance || []),
+    ...(runChecks.seo || []),
+  ];
+  return allResults.some((result) => result.status === 'fail');
+}
```

Comment:
That's wrong, you need to reliably return your results, you say it doesn't have any failures, just because u didn't run any checks.